vezba-stabla/src/StabloProgram.java

```diff
@@ -0,0 +1,203 @@
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+
+import org.svetovid.Svetovid;
+
+class Stablo {
+    private static class Cvor {
+    	Automobil info;
+    	Cvor levo;
+    	Cvor desno;
+    	
+    	public Cvor(Automobil info, Cvor levo, Cvor desno) {
+    		this.info = info;
+    		this.levo = levo;
+    		this.desno = desno;
+    	}
+    	
+    	public Cvor kopija() {
+    		return new Cvor(info, levo, desno);
+    	}
+    }
+
+    private Cvor koren;
+
+    // konstruktor za prazno stablo
+    public Stablo() {
+    	this.koren = null;
+    }
+    
+    public Stablo(Automobil auto) {
+    	this.koren = new Cvor(auto, null, null);
+    }
+    
+    protected Stablo(Cvor cvor) {
+    	this.koren = cvor;
+    }
+    
+    
+    // ------ dodati opisane metode ------------
+    
+    // ispisuje registracije svih automobila iz Novog Sada
+    public void ispisiNS() {
+    	System.out.println("Svi automobili iz Novog Sada: ");
+    	sviIzNs(koren);
+    }
+    
+    private void sviIzNs(Cvor cvor) {
+    	if (cvor == null) 
+    		return;
+    	
+    	if (cvor.info.getRegistracija().contains("NS")) {
+    		System.out.println(cvor.info.getRegistracija());
+    	}
+    	
+    	sviIzNs(cvor.levo);
+    	sviIzNs(cvor.desno);    	
+    }
+    
+    // vraca sumu kilometraza svih plavih automobila
+    public long presliPlavi() {
+    	Suma suma = new Suma();

```

Comment:
Ovo je okej i po pravilu se tako radi OOP ali za primitivnu radnju poput sumiranja intidžera u jednu promenljivu, nije ti potrebno da praviš klasu. Ne moraš prepravljati ovo ali obično se ne pravi klasa za te stvari jer dodaje bespotreban overhead na prostu operaciju koja nije dovoljno kompleksna da bi zaslužila apstrakciju što bi se reklo :).